src/core/mac/mac.cpp

```diff
@@ -2050,8 +2183,87 @@ uint8_t Mac::GetTimeIeOffset(const Frame &aFrame)
 }
 #endif
 
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+void Mac::SetCslChannel(uint8_t aChannel)
+{
+    VerifyOrExit(GetCslChannel() != aChannel, OT_NOOP);
+
+    mSubMac.SetCslChannel(aChannel);
+
+    if (IsCslEnabled())
+    {
+        Get<Mle::Mle>().ScheduleChildUpdateRequest();
+    }
+exit:
+    return;
+}
+
+void Mac::SetCslPeriod(uint16_t aPeriod)
+{
+    mSubMac.SetCslPeriod(aPeriod);
+    IgnoreError(Get<Radio>().EnableCsl(GetCslPeriod(), &Get<Mle::Mle>().GetParent().GetExtAddress()));
+
+    if (IsCslEnabled())
+    {
+        Get<Mle::Mle>().ScheduleChildUpdateRequest();
+    }
+
+    UpdateIdleMode();
+}
+
+void Mac::SetCslTimeout(uint32_t aTimeout)
+{
+    VerifyOrExit(GetCslTimeout() != aTimeout, OT_NOOP);
+
+    mSubMac.SetCslTimeout(aTimeout);
+
+    if (IsCslEnabled())
+    {
+        Get<Mle::Mle>().ScheduleChildUpdateRequest();
+    }
+
+exit:
+    return;
+}
+
+bool Mac::IsCslEnabled(void) const
+{
+    return (GetCslPeriod() > 0) && !GetRxOnWhenIdle() && Get<Mle::MleRouter>().IsChild() &&
+           Get<Mle::Mle>().GetParent().IsEnhancedKeepAliveSupported();
```

Comment:
```suggestion
           Get<DataPollSender>().GetParent();
```

Reply:
> Note that UpdateFrameControlField() calls IsCslEnabled() which wraps the parent version relative check already.
> And UpdateFrameControlField() only uses aNeighbor info for CSL_TRANSMITER to determine the 154 version according the child's sync status.

`IsCslEnabled` is only used by **SSED**.  It couldn't be used for a CSL transmitter. `aNeighbor` is used to determine that if the current frame sent is using csl transmission from a CSL transmitter to a CSL receiver.

Regarding the arrangement of the code(why passing `aNeighbor` instead of `aIfCsl...`), there was a discussion. The outcome is that in current arrangement, everything is only calculated only once.